### PR TITLE
Parameterizing trigger name rule name as well as triggers and actions inside rule

### PR DESCRIPTION
**Pull request by beemarie**

### CODE REVIEW

File: parsers/manifest_parser_test.go [hunk 1]
```diff
@@ -1683,6 +1683,39 @@ func TestPackageName_Env_Var(t *testing.T) {
...
+    mm := NewYAMLParser()
+    manifestfile := "../tests/dat/manifest_data_rule_env_var.yaml"
+    manifest, _ := mm.ParseManifest(manifestfile)
+    rules, err := mm.ComposeRulesFromAllPackages(manifest)
+    if err != nil {
+        assert.Fail(t, "Failed to compose rules")
+    }
+    packageName := "manifest1"
+
+    assert.Equal(t, 1, len(manifest.Packages[packageName].Rules), "Get trigger list failed.")
```
Comment (mrutkows) [inline parsers/manifest_parser_test.go]: Rule list?

File: parsers/yamlparser.go [hunk 2]
```diff
@@ -177,16 +177,17 @@ func (yaml *YAML) GetProject() Project {
 	return yaml.Application
 }
 
-func convertSinglePackageName(packageName string) string {
-	if len(packageName) != 0 {
-		packageNameEnv := wskenv.GetEnvVar(packageName)
-		if str, ok := packageNameEnv.(string); ok {
+
+func convertSingleName(theName string) string {
```
Comment (mrutkows) [inline parsers/yamlparser.go]: Can you move this function to the package wskenv please? (i.e., wskenv/environment.go )?

File: tests/dat/manifest_data_rule_env_var.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      license: Apache-2.0
+      actions:
+        first_action:
+          function: actions/hello.js
+      triggers:
+        $trigger_name:
+      rules:
+        $rule_name:
+          trigger: $trigger_name
+          action: $action_name
```
Comment (mrutkows) [inline tests/dat/manifest_data_rule_env_var.yaml]: $action_name replacement should be tested elsewhere? trigger-rule indeed belong together here.

Comment (beemarie) [reply tests/dat/manifest_data_rule_env_var.yaml]: hrmm, it's not the actual action name replacement. Just the ability to parameterize the action being connected in the rule.

**Pull request merged**